### PR TITLE
Pong game online/local

### PR DESCRIPTION
**Pull request by Neffi42**
- **create resetEvents function**
- **create WebSocket functions**
- **create game's ccs file**
- **create games app**
- **create pong page**
- **create Pong model**
- **add Pong to admin page**
- **create Pong game**

### CODE REVIEW

File: django/src/games/pong.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+            await self.close()
+            return
+        self.valid_consumer = True
+        await self.accept()
+        await self.channel_layer.group_add(self.group_name, self.channel_name)
+        await self.send_message("group", "game_join", {"user": self.scope["user"].id})
+        await self.send_message("client", "game_pad", {"game_pad": self.pad_n})
+
+    def initialize_game(self):
+        self.redis = redis.Redis(host="redis")
```
Comment (antoineverin) [inline django/src/games/pong.py:27]: can redis have to much connections?

Comment (Neffi42) [reply django/src/games/pong.py:27]: Need to investigate in the future...

File: django/src/games/pong.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+                await asyncio.sleep(fps)
+            except Exception as e:
+                print(f"Unexpected error in the loop from game {self.room_id}: {e}")
+                break
+
+    def get_winner(self):
+        if self.mode == "online":
+            return self.info.players[0] if self.info.score[0] > self.info.score[1] else self.info.players[1]
+        else:
+            return "Pad 1" if self.info.score[0] > self.info.score[1] else "Pad 2"
```
Comment (antoineverin) [inline django/src/games/pong.py:146]: subject require all players to have usernames

**Review by antoineverin — CHANGES_REQUESTED** [review]

**Review by croueAlec — CHANGES_REQUESTED** [review]
when a player leaves a game, the opponent's score is automatically set to the max score

File: django/src/games/pong.py (hunk 2 above)
Comment (Neffi42) [reply django/src/games/pong.py:147]: It's only for local games

**Review by antoineverin — APPROVED** [review]

**Review by croueAlec — APPROVED** [review]
approved 👍

**Pull request merged**